src/Robo/Plugin/Commands/DevelopmentModeCommands.php

```diff
@@ -77,12 +116,219 @@ public function devRefreshAll(
             $result = $this->devRefreshDrupal(
                 environmentType: LocalDevEnvironmentTypes::from($environmentType),
                 siteName: $siteName,
-                startLocalEnv: $startLocalDev,
             );
         }
         return $result;
     }
 
+    /**
+     * Refresh a site database in DDEV.
+     *
+     * @param string $siteName
+     *   The Drupal site name.
+     * @option db
+     *   Provide a path to a database dump to be used instead of downloading the latest dump.
+     */
+    public function databaseRefreshDdev(string $siteName = 'default', array $options = ['db' => '']): Result
+    {
+        // @todo: Update this method to not be DDEV specific.
+        $this->io()->title('DDEV database refresh.');
+
+        ['db' => $dbPath] = $options;
+        // Track whether a database path was provided by the user or not.
+        $dbPathProvidedByUser = $dbPath !== '';
+
+        if (!$dbPathProvidedByUser) {
+            $dbPath = $this->databaseDownload($siteName);
+        }
+
+        $this->io()->section("importing $siteName database.");
+        $this->say("Importing $dbPath");
+        $this->taskExec(LocalDevEnvironmentTypes::DDEV->value)
+            ->arg('import-db')
+            ->option('database', $siteName == 'default' ? 'db' : $siteName)
+            ->option('file', $dbPath)
+            ->run();
+
+        // If a database was downloaded as part of this process, delete it.
+        if (!$dbPathProvidedByUser) {
+            $this->deleteDatabase($dbPath);
+        }
+
+        return $this->drushDeployWith(
+            localEnvironmentType: LocalDevEnvironmentTypes::DDEV,
```

Comment:
Do we still need this? Seems like if we are defaulting to one local type, `drushDeployWith()` should always do a local deploy (ie currently ddev). Again, if we use an interface, base, and child pattern, this could be handled that way. Probably more of a refactor than desired for now.

Reply:
See above.